landscape/lib/network.py

```diff
@@ -10,129 +10,63 @@
 import errno
 import logging
 
-from twisted.python.compat import long, _PY3
+import netifaces
+from twisted.python.compat import long
 
-__all__ = ["get_active_device_info", "get_network_traffic", "is_64"]
+__all__ = ["get_active_device_info", "get_network_traffic"]
 
-# from header /usr/include/bits/ioctls.h
-SIOCGIFCONF = 0x8912
-SIOCGIFFLAGS = 0x8913
-SIOCGIFNETMASK = 0x891b
-SIOCGIFBRDADDR = 0x8919
-SIOCGIFADDR = 0x8915
-SIOCGIFHWADDR = 0x8927
 
+SIOCGIFFLAGS = 0x8913  # from header /usr/include/bits/ioctls.h
 SIOCETHTOOL = 0x8946  # As defined in include/uapi/linux/sockios.h
 ETHTOOL_GSET = 0x00000001  # Get status command.
 
-# struct definition from header /usr/include/net/if.h
-# the struct size varies according to the platform arch size
-# a minimal c program was used to determine the size of the
-# struct, standard headers removed for brevity.
-"""
-#include <linux/if.h>
-int main() {
-  printf("Size of struct %lu\n", sizeof(struct ifreq));
-}
-"""
-
-IF_STRUCT_SIZE_32 = 32
-IF_STRUCT_SIZE_64 = 40
-
 
 def is_64():
     """Returns C{True} if the platform is 64-bit, otherwise C{False}."""
     return struct.calcsize("l") == 8
 
 
-# initialize the struct size as per the machine's architecture
-IF_STRUCT_SIZE = is_64() and IF_STRUCT_SIZE_64 or IF_STRUCT_SIZE_32
-
-
-def get_active_interfaces(sock):
-    """Generator yields active network interface names.
-
-    @param sock: a socket instance.
-    """
-    max_interfaces = 128
+def get_active_interfaces():
+    for interface in netifaces.interfaces():
```

Comment:
Add a docstring please, calling out that this is a generator

Reply:
Updated docstring.